libpromises/evalfunction.c

```diff
@@ -4670,14 +4680,36 @@ static FnCallResult FnCallIPRange(EvalContext *ctx, ARG_UNUSED const Policy *pol
 
     for (const Item *ip = EvalContextGetIpAddresses(ctx); ip != NULL; ip = ip->next)
     {
-        if (FuzzySetMatch(range, VIPADDRESS) == 0 ||
-            FuzzySetMatch(range, ip->name) == 0)
+        Rlist *ifaces = finalargs->next;
+        // we match on VIPADDRESS iff no interfaces were requested
+        if (FuzzySetMatch(range, VIPADDRESS) == 0 && NULL == ifaces)
         {
-            return FnReturnContext(true);
+            Log(LOG_LEVEL_DEBUG, "%s: found range %s on hostip %s, no interface", fp->name, range, ip->name);
+            return FnReturnContextTrue();
+        }
+        else if (FuzzySetMatch(range, ip->name) == 0)
+        {
+            if (NULL == ifaces) // no interfaces requested
+            {
+                Log(LOG_LEVEL_DEBUG, "%s: found range %s on IP %s, any interface", fp->name, range, ip->name);
+                return FnReturnContextTrue();
+            }
+            else // check the specific interfaces given as arguments
+            {
+                for (; ifaces != NULL; ifaces = ifaces->next)
+                {
+                    Log(LOG_LEVEL_DEBUG, "%s: checking range %s on IP %s, interface %s", fp->name, range, ip->name, RlistScalarValue(ifaces));
+                    if (0 == strcmp(RlistScalarValue(ifaces), ip->classes))
```

Comment:
Lol! I guess we're being a bit paranoid, but yes, that sounds like the safest option.